crypto/rand/rand_unix.c

```diff
@@ -119,26 +133,92 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
 #   error "Seeding uses urandom but DEVRANDOM is not configured"
 #  endif
 
+#  if defined(__GLIBC__) && defined(__GLIBC_PREREQ)
+#   if __GLIBC_PREREQ(2, 25)
+#    define OPENSSL_HAVE_GETRANDOM
+#   endif
+#  endif
+
+#  if (defined(__FreeBSD__) && __FreeBSD__ >= 12)
```

Comment:
It's probably better to use ```__FreeBSD_version``` from ```sys/param.h```and check the actual value against a fine-grained constant, since people might be running a snapshot of 12-current that is too old to have this support.  Unfortunately, the addition of getrandom does not seem to appear in https://www.freebsd.org/doc/en/books/porters-handbook/versions.html so some detective work will be required...

Reply:
Oh, I didn't know 12 was already that old. I guess I could go for 1200061 which is the first version there after it's been merged.

Reply:
Yeah, something that calls itself 12 has existed since 11.0 was released.

Reply:
Pushed a commit for that. Any idea when the sysctl support was added so I can do the same for that?

Reply:
Current oldest reference I could find is:
https://lists.freebsd.org/pipermail/svn-src-all/2012-July/056019.html

Reply:
I seem to have found it: https://github.com/freebsd/freebsd/commit/3ef9d41401d2f9d9b32d2c476de4aee9b8049188